File: Python/0101-0150/111. Minimum Depth of Binary Tree.py
# https://leetcode.com/problems/minimum-depth-of-binary-tree/description/

class Solution:
    def minDepth(self, root: Optional[TreeNode]) -> int:
        if root == None:
            return 0
        if root.left == None and root.right == None:
            return 1

        left = self.minDepth(root.left) if (root.left != None) else 9999
        right = self.minDepth(root.right) if (root.right != None) else 9999

        return min(left, right)+1

File: Java/0101-0150/111. Minimum Depth of Binary Tree.java
// https://leetcode.com/problems/minimum-depth-of-binary-tree/description/

class Solution {
    public int minDepth(TreeNode root) {
        if(root == null) return 0;
        if(root.left == null && root.right == null) return 1;

        int left = (root.left != null) ? minDepth(root.left) : Integer.MAX_VALUE;
        int right = (root.right != null) ? minDepth(root.right) : Integer.MAX_VALUE;

        return Math.min(left, right)+1;
    }
}
